program-runtime/src/compute_budget.rs

```diff
@@ -219,15 +226,21 @@ impl ComputeBudget {
         }
 
         self.max_units = if default_units_per_instruction {
-            requested_units
-                .or_else(|| Some(num_instructions.saturating_mul(DEFAULT_UNITS as usize) as u64))
+            requested_units.or_else(|| {
+                Some(
+                    num_non_compute_budget_instructions.saturating_mul(DEFAULT_UNITS as usize)
+                        as u64,
+                )
+            })
         } else {
             requested_units
         }
         .unwrap_or(MAX_UNITS as u64)
         .min(MAX_UNITS as u64);
 
-        Ok(prioritization_fee.unwrap_or(0))
+        Ok(prioritization_fee
+            .map(|fee_type| PrioritizationFeeDetails::new(fee_type, self.max_units))
```

Comment:
Outside of the scope of this PR, but do we enforce a minimum `max_units` so that we can avoid people spamming unreasonable transactions into the buffer?

Seems like `0, 1, 2...minimum` low max_unit transactions should just be dropped from the buffer

Reply:
Agreed, responded on your other comment.